### ISSUE TITLE
Build fails when building in parallel

### ISSUE THREAD

**Issue by folker-kuhn** [piranhacloud/piranha#1948]
Building Piranha with mvn clean install -T8 results in errors. mvn clean install without -T8 works, but takes like 10 minutes in my system.

Can you fix the errors during the parallel build? Thanks!

**Comment by Thihup** [comment]
Hi @folker-kuhn!
We have some integration tests that currently can't run in parallel. What you could do is skip the tests and the Javadoc, as most of the build time is because of them. 
`mvn clean install -T8 -DskipTests -Dmaven.javadoc.skip=true`

**Comment by folker-kuhn** [comment]
Oh, that is not a bad idea, thanks!

I think this fixes my problem, but out of the box parallel would be good to help others, not? Why dont integration test can run in parallel? Can this be fixed?

**Comment by Thihup** [comment]
I did a quick look in what needs to be done:
- Use different ports for the HTTP tests (using a library like https://github.com/alexpanov/freeportfinder could help here)
- Add a dependency on Piranha Server and Micro on some tests (/test/server/pom.xml and /test/micro/pom.xml) to make sure they start only after the project has built it 

Unfortunately, we still need a single run to install the artifacts in the local Maven repo to Arquillian find the dependencies.

Other than that, it seems possible.

**Comment by arjantijms** [comment]
That sounds cool to do. Especially fixing the dependency parts has allowed the GlassFish project to build in parallel out of the box.

**Comment by mnriem** [comment]
Nice! What is the speedup you are seeing?

**Comment by Thihup** [comment]
@mnriem In my machine, a full build (`mvn clean install`) it went down from about 14 minutes to 2 minutes (using `mvn clean install -T8`).

Sometimes, however, we get a pretty confusing error, like this one: 
[cloud.piranha.test.faces.mojarra.Mojarra2Test.txt](https://github.com/piranhacloud/piranha/files/7403397/cloud.piranha.test.faces.mojarra.Mojarra2Test.txt)

It seems that it reads an invalid zip file, so I need to take a closer look. But otherwise, the improvement seems to be pretty good.

**Comment by arjantijms** [comment]
That looks pretty fantastic indeed. Building in parallel (without the tests) already gave me a huge speed boost here; 24 seconds (no tests, parallel) vs 7 minutes (tests, serial).

**Comment by mnriem** [comment]
My gut on the confusing error would be that say process A is trying to read the signature of a given JAR/ZIP and say process B is updating the same file ... you probably want to add some additional logging so it shows you which JAR/ZIP file is tripping you up?

**Comment by mnriem** [comment]
@Thihup Do you need any help here? If so let us know!

**Comment by Thihup** [comment]
@mnriem What I'm seeing is that when testing Piranha Micro, the Shrinkwrap Maven resolver changes something in the filesystem (folder ~/.m2) in the same instant as Maven is doing something else. I actually don't know how to solve it.
The changes in the PR already improves time using `mvn package`, but `mvn verify` / `mvn install` still fail.

**Comment by mnriem** [comment]
@arjantijms Any insight here? @Thihup do these changes require any changes to the current command line and or can they be committed regardless and for now only improve `mvn package` ?

**Comment by Thihup** [comment]
@mnriem The changes can be already committed. I didn't merged it yet because it didn't improve the `mvn install`, but we can merge it now and latter fix the `mvn install` 😉

**Comment by mnriem** [comment]
I suggest we merge it and open a new issue to address the remaining work later.
